core/middlewares/with-routes.ts

```diff
@@ -282,21 +289,36 @@ export const withRoutes: MiddlewareFactory = () => {
     };
 
     if (route?.redirect) {
+      // Only carry over query params if the fromPath does not have any, as Bigcommerce 301 redirects support matching by specific query params.
+      const fromPathHasParams = new URL(route.redirect.fromPath, request.url).search.length > 0;
+      const searchParams = fromPathHasParams ? '' : request.nextUrl.search;
```

Comment:
I try to lean on boolean variables name as little as possible. I get they make hard to read logic more readable, but they generally tend to only be used once or twice.
```suggestion
      // Only carry over query params if the fromPath does not have any, as Bigcommerce 301 redirects support matching by specific query params.
      const fromPathSearchParams = new URL(route.redirect.fromPath, request.url).search;
      const searchParams = fromPathSearchParams.length > 0 ? '' : request.nextUrl.search;
```